plugin-build/src/main/kotlin/io/sentry/android/gradle/instrumentation/logcat/LogcatInstrumentable.kt

```diff
@@ -0,0 +1,23 @@
+package io.sentry.android.gradle.instrumentation.logcat
+
+import com.android.build.api.instrumentation.ClassContext
+import io.sentry.android.gradle.instrumentation.ClassInstrumentable
+import io.sentry.android.gradle.instrumentation.SpanAddingClassVisitorFactory
+import io.sentry.android.gradle.instrumentation.util.isSentryClass
+import org.objectweb.asm.ClassVisitor
+
+class SentryLogcatInstrumentable(private val minLevel: LogcatLevel) :
+    ClassInstrumentable {
+
+    override fun getVisitor(
+        instrumentableContext: ClassContext,
+        apiVersion: Int,
+        originalVisitor: ClassVisitor,
+        parameters: SpanAddingClassVisitorFactory.SpanAddingParameters
+    ): ClassVisitor {
+        return LogcatClassVisitor(apiVersion, originalVisitor, minLevel)
+    }
+
+    override fun isInstrumentable(data: ClassContext) =
+        !data.isSentryClass() && !data.currentClassData.className.contains("SentryLogcatAdapter\$Companion")
```

Comment:
```suggestion
        !data.isSentryClass()
```
I think `isSentryClass()` should already cover it, because it checks if the class name starts with `io.sentry`, which `SentryLogcatAdapter` falls under. Or am I missing something?

Reply:
It will run into a recursion because`Log.d` calls `SentryLogcatAdapter.d` which again calls `Log.d` and never break out of it but I'll check again if it's the case.

Reply:
if you've tested it locally and put the adapter into the `io.sentry.samples.**` package, then indeed it'd go into a loop, because we exclude `io.sentry.samples.**` package from `isSentryClass` check

Reply:
Oh yeah, true, that might be the case, my first local tests were done adding it in the samples.